File: compress_and_decompress.java
public class Main {
    public static void main(String[] args) {
        String example1 = "aaabbbcccdeefgg";
        String example2 = "a3b3c3d1e2f1g2";
        String example3 = "";
        System.out.println("Compress aaabbbcccdeefgg: " + compressString(example1));
        System.out.println("Compress empty string: " + compressString(example3));
        System.out.println("Decompress a3b3c3d1e2f1g2: " + decompressString(example2));
        System.out.println("Decompress empty string: " + decompressString(example3));
    }

    static String compressString(String word) {
        String compressed = "";
        int num = 1;

        if (!word.isEmpty()) {
            for (int i = 0; i < word.length() - 1; i++) {
                if (word.charAt(i) == word.charAt(i + 1)) {
                    num += 1;
                } else {
                    compressed += word.charAt(i) + Integer.toString(num);
                    num = 1;
                }
            }
            compressed += word.charAt(word.length() - 1) + Integer.toString(num);  
        }

        return compressed;
    }

    static String decompressString(String word) {
        String decompressed = "";

        if (!word.isEmpty()) {
            for (int i = 0; i < word.length(); i += 2) {
                char c = word.charAt(i);
                int repeatCount = word.charAt(i + 1) - '0'; 
                decompressed += String.valueOf(c).repeat(repeatCount);
            }
        }

        return decompressed;    
    }
}

File: compress_and_decompress.py
def compress_string(word):
  compressed = ""
  num = 1
  
  if word:
    for i in range(0, len(word) - 1):
      if word[i] == word[i+1]:
        num += 1
      else:
        compressed += word[i] + str(num)
        num = 1
    
    compressed += word[-1] + str(num)
  
  return compressed

def decompress_string(word):
  decompressed = ""
  
  if word:
    for i in range(0, len(word), 2):
      decompressed += str(word[i] * int(word[i+1]))

  return decompressed

example1 = "aaabbbcccdeefgg"
example2 = "a3b3c3d1e2f1g2"
example3 = ""
print(f"Compress aaabbbcccdeefgg: {compress_string(example1)}")
print(f"Compress empty string: {compress_string(example3)}")
print(f"Decompress a3b3c3d1e2f1g2: {decompress_string(example2)}")
print(f"Decompress empty string: {decompress_string(example3)}")
